source/extensions/transport_sockets/proxy_protocol/proxy_protocol.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "envoy/config/core/v3/proxy_protocol.pb.h"
+#include "envoy/network/connection.h"
+#include "envoy/network/transport_socket.h"
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/logger.h"
+
+using envoy::config::core::v3::ProxyProtocolConfig_Version;
+
+namespace Envoy {
+namespace Extensions {
+namespace TransportSockets {
+namespace ProxyProtocol {
+
+class ProxyProtocolSocket : public Network::TransportSocket,
```

Comment:
totally optional, but what do you think of a base class that's a pure wrapper, and subclassing that to make it clear what's being overridden and what's passthrough.

Reply:
Yeah, seems like a good idea. Could potentially be used for the tap socket as well. I'll look into it.